### ISSUE TITLE
Container Tests

### ISSUE THREAD

**Issue by juddey** [infinitered/ignite#437]
I know we talked about this in passing in other PR's but I'm keen to start writing tests for my container components. 

Not sure where to start, but mocking out redux might be a good place. (?)

Once we have some general direction I can create a template for the ever growing list of generators.

Advice appreciated.

**Comment by GantMan** [comment]
Nice!

You can test a lot like you do with existing components... the redux portion.. well.  You can test that it's wrapped in redux.  But you won't be able to access the wrapped component without setting the flag in redux connect.  Interesting problem.

**Comment by juddey** [comment]
Yep. I started to implement component-like tests but once I threw in 
`import { connect } from react-redux` things fell apart spectacularly.

I've been doing some reading and theoretically you should be able to mount the component using `enzyme` and test through there, but that doesn't seem to [work either](https://github.com/airbnb/enzyme/issues/190).

I think it needs some more thinking on and reading - its late here so parking until... 🌤

**Comment by juddey** [comment]
So have been looking at this a bit further.

Exercising the google-fu has given me plenty of reading, but I found this the best:
http://redux.js.org/docs/recipes/WritingTests.html#connected-components

There are two types of test, one to test the rendering of the component, and the other to test with redux connected (you pass a custom store in that type of test). I'm just looking at the first.

The docs suggest exporting the class as well as the default, which I have done for the test under question, which is:

```
import React from 'react'
import { SelectChildren } from '../../App/Containers/SelectChildren'
import { shallow } from 'enzyme'

// Mock out libs here, just to keep things in one file.
jest.mock('ramda', () => {})
jest.mock('ramdasauce', () => {})
jest.mock('reduxsauce', () => {})

const wrapper = shallow(<SelectChildren />)

xit('component exists', () => {
  expect(wrapper.length).toBe(1) // exists
})
```
I get this sad panda: 

![screen shot 2016-10-31 at 11 42 49 am](https://cloud.githubusercontent.com/assets/7445602/19840647/a45aab32-9f5f-11e6-88ff-7f971f0209d9.png)

I tried to mock the createActions out (several ways), and no dice.

Suggestions welcome! 😖

**Comment by GantMan** [comment]
Well mocking out createActions isn't the issue.  It's whatever createActions is supposed to be a property of.  So some other import seems to be null.  Def a difficult test.  I'll def take a shot at it, but not today.  Today I hope to get back around to enabling maps 🗺

**Comment by juddey** [comment]
Hmm that's reduxsauce I think. Happy to wait for a push in the right direction. ✅

**Comment by juddey** [comment]
https://blog.hellojs.org/testing-react-native-and-redux-e5a71b99e178#.v91z4gp4k

**Comment by juddey** [comment]
Closing this issue. Turns out the mocking wasn't the problem - the mocks to ramda et al were not needed. I had a call to `R.map` (ie ramdas version of map), jest confused it with ES6's new `map` function, and things exploded as documented above.

I replaced the ramda map function with the ES6 map function (eg `object.map`) and all fixed.

Thinking about how I'll submit a container test for v2.0 of ignite.
